modules/library/xml/src/main/java/org/geotools/xml/SchemaFactory.java

```diff
@@ -133,7 +133,11 @@ private Map loadSchemas() {
                             new BufferedReader(new InputStreamReader(res.openStream(), "UTF-8"));
 
                     while (rd.ready()) {
-                        String factoryClassName = rd.readLine().trim();
+
+                        String factoryClassName = rd.readLine();
+                        // null check as per NP_IMMEDIATE_DEREFERENCE_OF_READLINE
+                        if (factoryClassName == null) break;
+                        factoryClassName = factoryClassName.trim();
```

Comment:
these changes seem unrelated to GEOT-6210

Reply:
They are, but best go in anyways. @imranrajjad had a Spotbugs report on that line which would break the QA build. It is bizzarre that it happens on his machine and not on mine, but we're using different OS, slightly different JDK versions, and there is a chance the eclipse compiler is involved too... In any case, the report is legit, readLine output should be checked for nulls and was not, so is the fix. Best to have the QA build work on as many platforms as possible.